### PR TITLE
chore(NODE-3719): adl spec compliance refactor

### PR DESCRIPTION
**Pull request by dariakp**
### Description
NODE-3719

#### What is changing?
Reorganizing ADL spec tests to clearly map to spec.

##### Is there new documentation needed for these changes?
N/A

#### What is the motivation for this change?
Auditing spec compliance

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const path = require('path');
+const {
+  TestRunnerContext,
+  gatherTestSuites,
+  generateTopologyTests
+} = require('../../tools/spec-runner');
+
+describe('Atlas Data Lake - spec', function () {
+  const testContext = new TestRunnerContext({
+    skipPrepareDatabase: true,
```
Comment (dariakp) [inline test/manual/atlas-data-lake-testing/atlas_data_lake_testing.spec.js:11]: The spec says that the runner setup is 

> The same as the CRUD Spec Test Runner Implementation, with one notable differences: the test runner for Atlas Data Lake Testing MUST NOT drop the collection and/or database under test. In contrast to other CRUD tests, which insert their own data fixtures into an empty collection, data for these tests is specified in the mongohoused configuration file.

Does this skipPrepareDatabase flag fully take care of that requirement?

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+'use strict';
+const { expect } = require('chai');
+const { withClient } = require('../../integration/shared');
+
+// TODO: These tests are not fully implemented per the spec
```
Comment (dariakp) [inline test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js:5]: Do we want a separate ticket for fixing this?

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+     * and cursor.ns fields, respectively. Destroy the cursor object and observe a CommandStartedEvent and CommandSucceededEvent for the killCursors command.
+     *
+     * Assert that the cursor ID and target namespace in the outgoing command match the values from the find command's CommandSucceededEvent.
+     * When matching the namespace, note that the killCursors field will contain the collection name and the database may be inferred from either
+     * the $db field or accessed via the CommandStartedEvent directly.
+     *
+     * Finally, assert that the killCursors CommandSucceededEvent indicates that the expected cursor was killed in the cursorsKilled field.
+     *
+     * Note: this test assumes that drivers only issue a killCursors command internally when destroying a cursor that may still exist on the server.
+     * If a driver constructs and issues killCursors commands in other ways (e.g. public API), this test MUST be adapted to test all such code paths.
```
Comment (dariakp) [inline test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js:24]: We are clearly not implementing the full description of the case correctly here, I think maybe this is testing one of the "other ways"? But in that case we still need to test the first part - the internal and the public codepaths, I assume. Could use help interpreting this line.

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  );
+
+  it(
+    '3a. Test that the driver can establish a connection with Atlas Data Lake with authentication. (SCRAM-SHA-1)',
+    /**
+     * For these tests, create a MongoClient using a valid connection string with SCRAM-SHA-1 and credentials
+     * from the drivers-evergreen-tools ADL configuration and execute a ping command.
+     */
+    withClient(
+      'mongodb://mhuser:pencil@localhost?authMechanism=SCRAM-SHA-1',
```
Comment (dariakp) [inline test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js:53]: Is this affected by our current (lack of) auth setup?  Also, should we be specifying the credentials differently?

**Review by dariakp — COMMENTED** [review]
Some open questions - not aiming to have a fully functional implementation, just enough info to note next steps

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js (hunk 4 above)
Comment (nbbeeken) [reply test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js:53]: ADL is setup to always run with auth enabled config file is here: https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/atlas_data_lake/config.yml

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.spec.js (hunk 1 above)
Comment (nbbeeken) [reply test/manual/atlas-data-lake-testing/atlas_data_lake_testing.spec.js:11]: I believe so: prepareDatabaseForSuite in https://github.com/mongodb/node-mongodb-native/blob/NODE-3795/fix-no-authProvider-for-default-error/test/tools/spec-runner/index.js#L201 is where the flag is used and you can see it returns early before any set up logic that would drop dbs/collections

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js (hunk 2 above)
Comment (nbbeeken) [reply test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js:5]: I think the best approach would be yes lets file an improvement ticket and keep sorting through our compliance

File: test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js (hunk 3 above)
Comment (nbbeeken) [reply test/manual/atlas-data-lake-testing/atlas_data_lake_testing.prose.js:24]: Yea I'm not seeing the logic to issue a query to have the outstanding cursor remain open, and then subsequently issue killCursors. I'm not sure but our cursor.close() may issue a killCursors, so we need to test doing it this way with a db.command and see what happens with a cursor.close call as well.

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**